toodlext/src/main.rs

```diff
@@ -0,0 +1,355 @@
+#[macro_use]
+extern crate serde_derive;
+
+extern crate byteorder;
+extern crate serde;
+extern crate serde_json;
+extern crate toodle;
+
+use std::collections::HashSet;
+use std::io::{self, Read, StdinLock, StdoutLock, Write};
+use std::iter::FromIterator;
+
+use byteorder::{NativeEndian, ReadBytesExt, WriteBytesExt};
+use toodle::{Timespec, Toodle, Uuid};
+use toodle::items::Item;
+use toodle::labels::Label;
+
+#[derive(Serialize, Debug)]
+enum Error {
+    IOError,
+    BadJSON,
+    BadRequest,
+    ItemNotFound,
+    UpdateItemFailed,
+    UpdateLabelsFailed,
+    NotImplemented,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct LabelInfo {
+    name: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    color: Option<String>,
+}
+
+impl LabelInfo {
+    fn from_label(label: &Label) -> LabelInfo {
+        LabelInfo {
+            name: label.name.clone(),
+            color: Some(label.color.clone()),
+        }
+    }
+
+    fn to_label(&self) -> Label {
+        Label {
+            id: None,
+            name: self.name.clone(),
+            color: self.color.clone().unwrap_or_default(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct ItemInfo {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    uuid: Option<String>,
+
+    name: String,
+
+    #[serde(rename = "dueDate", skip_serializing_if = "Option::is_none")]
+    due_date: Option<i64>,
+
+    #[serde(rename = "completionDate", skip_serializing_if = "Option::is_none")]
+    completion_date: Option<i64>,
+
+    labels: Option<Vec<LabelInfo>>,
+}
+
+impl ItemInfo {
```

Comment:
Take a look at the `Into` trait. By implementing `Into` for a type then you get `From` for free, which means you don't need to have both the to and from functions. https://doc.rust-lang.org/std/convert/trait.Into.html